joomla/joomla-cms
# [5.0] Remove utf8mb4 conversion

Pull Request for Issue # .

### Summary of Changes

This pull request (PR) removes all code related to the utf8mb4 conversion.

Joomla 4.0 and later require a minimum database version for MySQL or MariaDB which supports utf8mb4 and so would not run on databases which do not support it.

But as a result of a sever migration when updating from 3.10 to 4.x, it could still be that the database tables of the core haven't been converted to utf8mb4 yet, so the update to 4.x had to make that conversion.

Given that people have to be on a Joomla 4.x before they can update to 5.0, so updates from 3.10.x to 5.0 will not be possible, it is granted that the database has been converted in past when they had updated from 3.10.x to 4.y.

So for updating to 5.0 all code and SQL scripts related to that conversion are obsolete.

### Testing Instructions

The tests make only sense with MySQL or MariaDB databases but not with PostgreSQL.

Update any Joomla 4.x version to the custom update URL or the update package which have been built by drone for this PR and make sure that the update works.

After that, make sure that the database checker ("System -> Maintenance -> Database", i.e. `administrator/index.php?option=com_installer&view=database`) still works.

### Actual result BEFORE applying this Pull Request

Obsolete, unused code and SQL scripts for utf8mb4 conversion of MySQL or MariaDB databases.

### Expected result AFTER applying this Pull Request

No obsolete, unused code and SQL scripts for utf8mb4 conversion of MySQL or MariaDB databases.

Updating from any Joomla 4.x version works as well as without this PR.

The database checker works as well as without this PR.

### Documentation Changes Required

None.
